submission/persistence-filesystem/src/main/kotlin/ac/uk/ebi/biostd/persistence/filesystem/nfs/NfsFilesService.kt

```diff
@@ -33,11 +33,13 @@ class NfsFilesService(
     }
 
     private fun persistNfsFile(sub: ExtSubmission, file: NfsFile): ExtFile {
-        val permissions = sub.permissions()
-        val subFile = getSubFile(sub, permissions, file.relPath)
-        if (subFile.notExist()) copyOrReplaceFile(file.file, subFile, permissions)
+        return synchronized(this) {
```

Comment:
why this need to be synchronized?

Reply:
Because the operation to getOrCreate the submission folders is not thread safe

Reply:
I think the problem with this functions are.

1. In FileUtils
```
    fun createFolderIfNotExist(file: Path, permissions: Set<PosixFilePermission>) {
        if (exists(file).not()) createDirectories(file, permissions)
    }
``` 
the problem is that after function check the file does not exists it try to created but other function already did it. I think we can solve this code easily by  ignoring FileAlreadyExists exception in that way function will be threadsafe. Not the cleanest solution but given JDK does not offer createIFNotExists atomic (threadsafe) function is the only thing we can do. 

2. In FileUtils
```
    private fun createDirectory(path: Path, permissions: Set<PosixFilePermission>) {
        Files.createDirectory(path)
        Files.setPosixFilePermissions(path, permissions)
    }
```
This again assume directory does not exists so we can apply the same fix as in 1.


In that way we can remove both syncronization (which will not work when multiple JVM or backends) and do the method threadsafe.